presto-hive/src/test/java/com/facebook/presto/hive/TestHiveSplitScheduling.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.spi.eventlistener.EventListener;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import static com.facebook.presto.hive.HiveSessionProperties.DYNAMIC_SPLIT_SIZES_ENABLED;
+import static io.airlift.tpch.TpchTable.ORDERS;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+@Test(singleThreaded = true)
+public class TestHiveSplitScheduling
+        extends AbstractTestQueryFramework
+{
+    @Override
+    public QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return HiveQueryRunner.createQueryRunner(
+                ImmutableList.of(ORDERS),
+                ImmutableMap.of(),
+                "sql-standard",
+                ImmutableMap.of("hive.max-initial-split-size", "1kB"),
+                Optional.empty());
+    }
+
+    @Test
+    public void testDynamicSplits()
+    {
+        try {
+            getQueryRunner().execute("CREATE TABLE test_orders WITH (partitioned_by = ARRAY['ds', 'ts']) AS " +
+                    "SELECT orderkey, orderpriority, comment, custkey, '2020-09-01' as ds, '00:01' as ts FROM orders WHERE orderkey < 1000 " +
+                    "UNION ALL " +
+                    "SELECT orderkey, orderpriority, comment, custkey, '2020-09-02' as ds, '00:02' as ts FROM orders WHERE orderkey >= 1000 AND orderkey < 2000");
+
+            TestHiveEventListenerPlugin.TestingHiveEventListener eventListener = getEventListener();
+
+            // Wait for previous events to finish
+            Thread.sleep(2000);
```

Comment:
Is there a way to avoid explicit sleep calls?

Reply:
I found that event manager is synchronous in these tests, so we can just remove all sleep calls